Sender/BMS_AttributeDataSender.c

```diff
@@ -0,0 +1,91 @@
+  
+/**********************************************
+Include header files
+***********************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "BMS_AttributeDataSender.h"
+  
+/**********************************************
+Variable declaration
+***********************************************/
+OperationMode ReadStatus, WriteStatus;
+float Temperature[]={};
+float ChargeRate[]={};
+int BufferArraysize=0;
+
+/*********************************************
+Function definitions
+**********************************************/
+
+/********************************************************************************
+ * Function: BMS_Readfromdatafile
+ 
+ * Description: This function reads the BMS parameter readings from the suitable 
+   data log file and stores into a buffer
+   
+ * input: void
+ 
+ * returns: Read status is Success(True) if the data is successfully read from the file.
+ *********************************************************************************/
+OperationMode BMS_Readfromdatafile()
+{
+  FILE *BMS_datafile;
+  int line=1;
+ ReadStatus= Failure;
+ 
+	
```

Comment:
This function opens the file, reads it, and iterates through it. When we make it release-worthy, we will need to include things like input-file corruption, etc. The complexity of this function will keep increasing. It will keep getting harder to test.